xGame/Java/xgame/src/main/java/com/xgame/service/engine/King.java

```diff
@@ -38,15 +38,15 @@ public String toString() {
 				if(column + o[0] >= 0 && column + o[0] <= 7 && row + o[1] >= 0 && row + o[1] <= 7) {
 			        move = toPosition(column + o[0], row + o[1]);
 					adjPiece = board.getPiece(move);
-					if(adjPiece == null || (adjPiece != null && adjPiece.getColor() != this.color)){
+					if((adjPiece == null || (adjPiece != null && adjPiece.getColor() != this.color)) 
+							&& !board.isThreatened(move, (color == Color.WHITE) ? Color.BLACK : Color.WHITE)){

```

Comment:
this produces a bug in certain cases when the board is determining if the king is in check.